sos/report/plugins/maas.py

```diff
@@ -8,133 +8,177 @@
-        # MAAS 3.5 deb:
         'maas-temporal',
-        'maas-apiserver',
-        'maas-agent',
-        # For the pre-3.5 snap:
+        'maas-temporal-worker',
         'snap.maas.supervisor',
-        # MAAS 3.5 snap uses `snap.maas.pebble` service, but it's not
-        # included here to prevent automatic journald log collection.
+        'snap.maas.pebble',
     )
 
-    option_list = [
-        PluginOpt('profile-name', default='', val_type=str,
-                  desc='Name of the remote API'),
-        PluginOpt('url', default='', val_type=str,
-                  desc='URL of the remote API'),
-        PluginOpt('credentials', default='', val_type=str,
-                  desc='Credentials, or the API key')
-    ]
-
-    is_snap = False
-
-    def _has_login_options(self):
-        return self.get_option("url") and self.get_option("credentials") \
-            and self.get_option("profile-name")
-
-    def _remote_api_login(self):
-        ret = self.exec_cmd(
-            f"maas login {self.get_option('profile-name')} "
-            f"{self.get_option('url')} {self.get_option('credentials')}"
-        )
-
-        return ret['status'] == 0
-
     def _is_snap_installed(self):
         maas_pkg = self.policy.package_manager.pkg_by_name('maas')
         if maas_pkg:
             return maas_pkg['pkg_manager'] == 'snap'
         return False
 
-    def setup(self):
-        self.is_snap = self._is_snap_installed()
-        if self.is_snap:
-            self.add_cmd_output([
-                'snap info maas',
-                'maas status'
-            ])
-
-            if self.is_service("snap.maas.pebble"):
-                # Because `snap.maas.pebble` is not in the services
-                # tuple to prevent timeouts caused by log collection,
-                # service status and logs are collected here.
-                self.add_service_status("snap.maas.pebble")
-                since = self.get_option("since") or "-1days"
-                self.add_journal(units="snap.maas.pebble", since=since)
-
-            # Don't send secrets
-            self.add_forbidden_path([
-                "/var/snap/maas/current/bind/session.key",
-                "/var/snap/maas/current/http/certs/regiond-proxy-key.pem",
-            ])
+    def _get_machines_syslog(self, dir):
+        if not os.path.exists(dir):
+            return []
+
+        # Machine messages are collected with syslog and are stored under:
+        #   $template "{{log_dir}}/rsyslog/%HOSTNAME%/%$YEAR%-%$MONTH%-%$DAY%"
+        # Collect only the most recent "%$YEAR%-%$MONTH%-%$DAY%"
+        # for each "%HOSTNAME%".
+        recent = []
+        for host_dir in os.listdir(dir):
+            host_path = os.path.join(dir, host_dir)
+            if not os.path.isdir(host_path):
+                continue
+
+            subdirs = [
+                os.path.join(host_path, d)
+                for d in os.listdir(host_path)
+                if os.path.isdir(host_path)
+            ]
```

Comment:
`Plugin` includes wrappers for these path methods, e.g. `Plugin.path_join()`, that are preferred because they will account for any `sysroot` handed to sos.

Reply:
Oh, TIL. Thanks!